### DISCUSSION TITLE
How to set null when withDefault are there on parseAsBoolean ?

### DISCUSSION THREAD

**Discussion by Kavan72** [47ng/nuqs#564] [Q&A] [answered]
I've state 
```
const LeadFilterType = {
  status: parseAsBoolean.withDefault(true)
}
```
and i want to set null on this, how can i achive this ?

**Comment by franky47** [comment, marked as answer]
`null` has a particular meaning in `nuqs`, it means _"there is no valid value in the URL for this key"_. 

When reading the state from the URL, this means if there is no key=value pair in the URL for this key, or if the parser fails to produce a correct JS value from the serialised value (eg, here for a boolean: `"not-a-boolean" -> null`.
Adding a default value on top uses the nullish coalescing operator (`??`) to transform that null into the default value.

When setting state, passing `null` means _"please remove the key=value pair from the URL"_. Then the read part described above kicks in.

So having a value that can be `T | null` (in your case, `bool | null`) is definitely possible: **don't set a default value**.
